driver/controller.go

```diff
@@ -0,0 +1,134 @@
+package driver
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/Juniper/contrail-go-api"
+	"github.com/Juniper/contrail-go-api/types"
+)
+
+const (
+	DomainName = "default-domain"
+)
+
+type Info struct {
+}
+
+type Controller struct {
+	ApiClient contrail.ApiClient
+}
+
+func NewController(ip string, port int) *Controller {
+	client := &Controller{}
+	client.ApiClient = contrail.NewClient(ip, port)
+	return client
+}
+
+func (c *Controller) GetNetwork(tenantName, networkName string) (*types.VirtualNetwork,
+	error) {
+	name := fmt.Sprintf("%s:%s:%s", DomainName, tenantName, networkName)
+	net, err := types.VirtualNetworkByName(c.ApiClient, name)
+	if err != nil {
+		return nil, err
+	}
+	return net, nil
+}
+
+func (c *Controller) GetDefaultGatewayIp(net *types.VirtualNetwork) (string, error) {
+	ipamReferences, err := net.GetNetworkIpamRefs()
+	if err != nil {
+		return "", err
+	}
+	if len(ipamReferences) == 0 {
+		return "", errors.New("Ipam references list is empty")
+	}
+	attribute := ipamReferences[0].Attr
+	ipamSubnets := attribute.(types.VnSubnetsType).IpamSubnets
+	if len(ipamSubnets) == 0 {
+		return "", errors.New("Ipam subnets list is empty")
+	}
+	gw := ipamSubnets[0].DefaultGateway
```

Comment:
We are the ones creating (and "managing") the network, and we don't need multiple subnets or ipamrefs